viadot/task_utils.py

```diff
@@ -599,3 +597,63 @@ def adls_bulk_upload(
             sp_credentials_secret=adls_sp_credentials_secret,
             overwrite=adls_overwrite,
         )
+
+
+@task(timeout=3600)
+def anonymize_df(
+    df: pd.DataFrame,
+    columns: List[str],
+    method: Literal["mask", "hash"] = "mask",
+    value: str = "***",
+    date_column: str = None,
+    days: int = None,
+) -> pd.DataFrame:
+    """
+    Function that anonymize data in the dataframe in selected columns.
+    It is possible to specify the condtition, for which data older than specified number of days will be anonymized.
+
+    Args:
+        df (pd.DataFrame): Dataframe with data to anonymize.
+        columns (List[str]): List of columns to anonymize.
+        method (Literal["mask", "hash"]): Method of anonymizing data. "mask" -> replace the data with "value" arg.
+            "hash" -> replace the data with the hash value of an object (using `hash()` method). Defaults to "mask".
+        value (str, optional): Value to replace the data. Defaults to "***".
+        date_column (str, optional): _description_. Defaults to None.
+        days (int, optional): The number of days beyond which we want to anonymize the data, e.g. older that 2 years can be: 2*365. Defaults to None.
+
+    Raises:
+        ValueError: If method or columns not found.
+
+    Returns:
+        pd.DataFrame: Operational dataframe with anonymized data.
+    """
+    if type(columns) is not list:
+        columns = columns.split(",")
+    if all(col in df.columns for col in columns) == False:
+        raise ValueError(
+            f"{columns} not found in dataframe. Provide proper column names."
```

Comment:
The user passes list of columns as an argument. If one of the col_names is wrong, he needs to pass the full list once again. Should I change only the ValueError message to be more like 
`f"At least one of the following columns is not found in dataframe: {columns}. Provide proper column names."`?

Reply:
Yes, it will be probably better message because we don't know where is the problem. We can check and list 'wrong' columns. Up to you. But this message is also ok.